### PR TITLE
added documentation

### LINKED ISSUE #7541: [C/CPP]: Flatten a Binary Tree
### Description

I would like to provide documentation on "Flatten a Binary Tree to LinkedList"

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by areenoverclouds**
<hr>

## Description 📜

I have added the documentation on Flatten a Binary Tree problem. 
Fixes #7541 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by deeqakkk — APPROVED** [review]
@Saviour1001 Do merge this ✅

**Comment by deeqakkk** [comment]
@areenoverclouds while making a PR, do name it according to your PR like what your PR is about or which issue it solves.

File: C_CPP/Data Structures/Trees/readme.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
 ## Trees
 
-- [Red Black Trees](RedBlackTrees)
\ No newline at end of file
+- [Red Black Trees](RedBlackTrees)
+- [Flatten Binary Tree)(FlattenBinaryTree.md)
```
Comment (Saviour1001) [inline C_CPP/Data Structures/Trees/readme.md:4]:
```suggestion
- [Flatten Binary Tree](FlattenBinaryTree.md)
```

**Comment by areenoverclouds** [comment]
> @areenoverclouds while making a PR, do name it according to your PR like what your PR is about or which issue it solves.

Am I supposed to add the issue number in the documentation?

**Comment by Saviour1001** [comment]
> Am I supposed to add the issue number in the documentation?
No , but it's just a good practice that your PR title should be to the point. Take care next time.

**Pull request merged**